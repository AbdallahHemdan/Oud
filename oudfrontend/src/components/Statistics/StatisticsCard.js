import React, { Component } from 'react'
import CountUp from "react-countup";
import { subUrl } from "./../../config/environment"

/**
 * @author AbdallahHemdan
 * @component
 * @description Statistics card is component to render the statistics card with all its statistics
 */
class StatisticsCard extends Component {

  /**
   * @function
   * @name render
   * @description Render Statistics cars
   * @returns {JSX} Component for App
   */

  render() {
    let {
      /**
       * @type {String}
       * #@description name of the song or the album of the current artist
       */
      name,
      /**
       * @type {String}
       * #@description image of the song or the album of the current artist
       */
      img,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album
       */
      listeners,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album
       */
      likes,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album per day
       */
      listenersPerDay,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album per week
       */
      listenersPerWeek,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album per month
       */
      listenersPerMonth,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album per day
       */
      likesPerDay,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album per week
       */
      likesPerWeek,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album per month
       */
      likesPerMonth
    } = this.props.data;
    return (
      <>
        <h5 className="item-name"
          data-testid="item-name"
        >{name}</h5>
        <img src={img} alt="song" className="card"
          data-testid="item-image"
        />
        <div className="statistics-data"
          data-testid="statistics-data-container">
          <h5 data-testid="listeners">Listeners : <span className="stat-number">{listeners}</span></h5>
          <h5 data-testid="likes">Likes : <span className="stat-number">{likes}</span></h5>
          <h5 data-testid="listeners-per-day">Listeners per day : <span className="stat-number">{listenersPerDay}</span></h5>
          <h5 data-testid="listeners-per-week">Listeners per week : <span className="stat-number">{listenersPerWeek}</span></h5>
          <h5 data-testid="listeners-per-month">Listeners per month : <span className="stat-number">{listenersPerMonth}</span></h5>
          <h5 data-testid="likes-per-day">Likes per day : <span className="stat-number">{likesPerDay}</span></h5>
          <h5 data-testid="likes-per-week">Likes per week : <span className="stat-number">{likesPerWeek}</span></h5>
          <h5 data-testid="likes-per-month">Likes per month : <span className="stat-number">{likesPerMonth}</span></h5>
        </div>
      </>
    )
  }
}

export default StatisticsCard
