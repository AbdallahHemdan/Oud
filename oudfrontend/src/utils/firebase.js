import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
