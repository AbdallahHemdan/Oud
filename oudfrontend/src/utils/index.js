import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
