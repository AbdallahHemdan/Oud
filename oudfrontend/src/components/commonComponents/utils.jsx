import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}