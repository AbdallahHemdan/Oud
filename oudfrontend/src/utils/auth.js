import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
