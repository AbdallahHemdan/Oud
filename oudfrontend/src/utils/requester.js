import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
