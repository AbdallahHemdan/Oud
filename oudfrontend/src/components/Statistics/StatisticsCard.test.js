import React from "react";
import { shallow } from 'enzyme';
import { findByTestAttr } from "./../../utils/index"
import renderer from 'react-test-renderer';
import StatisticsCard from './StatisticsCard';


const setup = (props = {}) => {
  const component = shallow(
    <StatisticsCard {...props} />
  )
  return component;
};

let data = {
  img: "Hemdan",
  name: "Hemdan",
  listeners: 1521,
  likes: 1521,
  listenersPerDay: 1521,
  listenersPerWeek: 1521,
  listenersPerMonth: 1521,
  likesPerDay: 1521,
  likesPerWeek: 1521,
  likesPerMonth: 1521
}
describe('StatisticsCard page Component', () => {
  let component;
  beforeEach(() => {
    component = setup({ data });
  })

  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "item-name");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "item-image");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "statistics-data-container");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "listeners");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "likes");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "listeners-per-day");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "listeners-per-week");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "listeners-per-month");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "likes-per-day");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "likes-per-week");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "likes-per-month");
    expect(wrapper.length).toBe(1);
  });

  describe('snapshot test', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<StatisticsCard data={data} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});