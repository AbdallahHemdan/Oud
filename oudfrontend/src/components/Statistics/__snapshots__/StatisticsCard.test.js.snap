// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsCard page Component snapshot test renders correctly 1`] = `
Array [
  <h5
    className="item-name"
    data-testid="item-name"
  >
    Hemdan
  </h5>,
  <img
    alt="song"
    className="card"
    data-testid="item-image"
    src="Hemdan"
  />,
  <div
    className="statistics-data"
    data-testid="statistics-data-container"
  >
    <h5
      data-testid="listeners"
    >
      Listeners : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="likes"
    >
      Likes : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="listeners-per-day"
    >
      Listeners per day : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="listeners-per-week"
    >
      Listeners per week : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="listeners-per-month"
    >
      Listeners per month : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="likes-per-day"
    >
      Likes per day : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="likes-per-week"
    >
      Likes per week : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
    <h5
      data-testid="likes-per-month"
    >
      Likes per month : 
      <span
        className="stat-number"
      >
        1521
      </span>
    </h5>
  </div>,
]
`;
